app/controllers/updateRegisteredDetails/WarehouseDetailsController.scala

```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.updateRegisteredDetails
+
+import utilities.GenericLogger
+import controllers.ControllerHelper
+import controllers.actions._
+import forms.updateRegisteredDetails.WarehouseDetailsFormProvider
+
+import javax.inject.Inject
+import models.{Mode, Warehouse}
+import pages.updateRegisteredDetails.WarehouseDetailsPage
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.SessionService
+import views.html.updateRegisteredDetails.WarehouseDetailsView
+import views.summary.updateRegisteredDetails.WarehouseDetailsSummary
+import handlers.ErrorHandler
+import models.backend.UkAddress
+
+import scala.concurrent.{ExecutionContext, Future}
+import navigation._
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
+import viewmodels.govuk.SummaryListFluency
+
+class WarehouseDetailsController @Inject()(
+                                       override val messagesApi: MessagesApi,
+                                       val sessionService: SessionService,
+                                       val navigator: NavigatorForUpdateRegisteredDetails,
+                                       identify: IdentifierAction,
+                                       getData: DataRetrievalAction,
+                                       requireData: DataRequiredAction,
+                                       formProvider: WarehouseDetailsFormProvider,
+                                       val controllerComponents: MessagesControllerComponents,
+                                       view: WarehouseDetailsView,
+                                       val genericLogger: GenericLogger,
+                                       val errorHandler: ErrorHandler
+                                     )(implicit ec: ExecutionContext) extends ControllerHelper with SummaryListFluency {
+
+  val form = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+
+      val preparedForm = request.userAnswers.get(WarehouseDetailsPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      val warehouse = request.userAnswers.warehouseList
+
+      val message: Option[SummaryList] = warehouse match {
+        case warehouseList  if !warehouseList.isEmpty => Some(SummaryListViewModel(
+          rows = WarehouseDetailsSummary.row2(warehouseList)
+        ))
+        case warehouseList if warehouseList.isEmpty => None
+      }
+
+      Ok(view(preparedForm, mode, message))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val Warehouse = request.userAnswers.warehouseList
+
+      val message: Option[SummaryList] = Warehouse match {
```

Comment:
perhaps call this summary list not message

also surely to be more readable this could be
val summaryList: Option[SummaryList] = warehouse.map(SummaryListViewModel(
          rows = WarehouseDetailsSummary.row2(warehouseList))
        )).

Reply:
The map on the map didn't work I have renamed it, the reason I have made it return a none is so I can show a message based on if there wasn't any warehouses in the list and show a summary list if there was a warehouse in the map